src/subcommand/server.rs

```diff
@@ -1742,6 +1743,17 @@ impl Server {
     })
   }
 
+  async fn charms() -> ServerResult {
```

Comment:
This can be:

```
  async fn charms() -> ServerResult<Json<BTreeMap<Charm, &'static str>>> {
```

`charm` implements serialize by delegating to `Display`, which is what is used in `to_string`.